src/util/util.ts

```diff
@@ -191,3 +191,23 @@ export function addArrayOfNumbers(arr: number[]): number {
 export function JSONClone<O>(object: O): O {
 	return JSON.parse(JSON.stringify(object));
 }
+
+/**
+ * @function
+ * @description Deep clone a class instance.
+ * @param instance The class instance you want to clone.
+ * @returns A new cloned instance.
+ */
+/*eslint-disable */
```

Comment:
dont do this, just disable return type checking, or better yet try to use a template type for the function since its input type is going to be the same as its return type.

and place this function in its own file